### ISSUE TITLE
Issues trying to run with tox poetry helpers

### ISSUE THREAD

**Issue by cburgess** [mtkennerly/poetry-dynamic-versioning#40]
Greeting,

First thanks for much for this plugin. This is one the main thing keeping us on setuptools+pbr.

I've gotten the plugin install and configure and working such that we can use poetry build and get package version we want. The problem is when I try and run our unit tests via tox. With `poetry-dynamic-versioning` disabled our unit tests work no problem. With `poetry-dynamic-versioning` enabled though we get the following error.

```
cfb@sandman:pistol.poetry/> venv-pistol/bin/tox -e py3
.package create: /Users/cfb/cisco/git/de/pistol.poetry/tox/py3
.package installdeps: poetry-core>=1.0.0, poetry-dynamic-versioning
py3 recreate: /Users/cfb/cisco/git/de/pistol.poetry/tox/py3
_______________________________________________________________ summary _______________________________________________________________
  py3: commands succeeded
  congratulations :)
Traceback (most recent call last):
  File "/Users/cfb/cisco/git/de/pistol.poetry/venv-pistol/bin/tox", line 8, in <module>
    sys.exit(cmdline())
  File "/Users/cfb/cisco/git/de/pistol.poetry/venv-pistol/lib/python3.9/site-packages/tox/session/__init__.py", line 44, in cmdline
    main(args)
  File "/Users/cfb/cisco/git/de/pistol.poetry/venv-pistol/lib/python3.9/site-packages/tox/session/__init__.py", line 69, in main
    exit_code = session.runcommand()
  File "/Users/cfb/cisco/git/de/pistol.poetry/venv-pistol/lib/python3.9/site-packages/tox/session/__init__.py", line 197, in runcommand
    return self.subcommand_test()
  File "/Users/cfb/cisco/git/de/pistol.poetry/venv-pistol/lib/python3.9/site-packages/tox/session/__init__.py", line 225, in subcommand_test
    run_sequential(self.config, self.venv_dict)
  File "/Users/cfb/cisco/git/de/pistol.poetry/venv-pistol/lib/python3.9/site-packages/tox/session/commands/run/sequential.py", line 9, in run_sequential
    if venv.setupenv():
  File "/Users/cfb/cisco/git/de/pistol.poetry/venv-pistol/lib/python3.9/site-packages/tox/venv.py", line 628, in setupenv
    status = self.update(action=action)
  File "/Users/cfb/cisco/git/de/pistol.poetry/venv-pistol/lib/python3.9/site-packages/tox/venv.py", line 275, in update
    self.hook.tox_testenv_install_deps(action=action, venv=self)
  File "/Users/cfb/cisco/git/de/pistol.poetry/venv-pistol/lib/python3.9/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/Users/cfb/cisco/git/de/pistol.poetry/venv-pistol/lib/python3.9/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/Users/cfb/cisco/git/de/pistol.poetry/venv-pistol/lib/python3.9/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
  File "/Users/cfb/cisco/git/de/pistol.poetry/venv-pistol/lib/python3.9/site-packages/pluggy/callers.py", line 208, in _multicall
    return outcome.get_result()
  File "/Users/cfb/cisco/git/de/pistol.poetry/venv-pistol/lib/python3.9/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/Users/cfb/cisco/git/de/pistol.poetry/venv-pistol/lib/python3.9/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/Users/cfb/cisco/git/de/pistol.poetry/venv-pistol/lib/python3.9/site-packages/tox_poetry_installer/hooks.py", line 71, in tox_testenv_install_deps
    poetry = utilities.check_preconditions(venv, action)
  File "/Users/cfb/cisco/git/de/pistol.poetry/venv-pistol/lib/python3.9/site-packages/tox_poetry_installer/utilities.py", line 49, in check_preconditions
    return _poetry.Factory().create_poetry(venv.envconfig.config.toxinidir)
  File "/Users/cfb/cisco/git/de/pistol.poetry/venv-pistol/lib/python3.9/site-packages/poetry_dynamic_versioning/__init__.py", line 296, in alt_poetry_create
    instance = original_poetry_create(cls, *args, **kwargs)
  File "/Users/cfb/cisco/git/de/pistol.poetry/venv-pistol/lib/python3.9/site-packages/poetry/factory.py", line 33, in create_poetry
    base_poetry = super(Factory, self).create_poetry(cwd)
  File "/Users/cfb/cisco/git/de/pistol.poetry/venv-pistol/lib/python3.9/site-packages/poetry_dynamic_versioning/__init__.py", line 326, in alt_poetry_create
    config = get_config(cwd)
  File "/Users/cfb/cisco/git/de/pistol.poetry/venv-pistol/lib/python3.9/site-packages/poetry_dynamic_versioning/__init__.py", line 134, in get_config
    pyproject_path = _get_pyproject_path(start)
  File "/Users/cfb/cisco/git/de/pistol.poetry/venv-pistol/lib/python3.9/site-packages/poetry_dynamic_versioning/__init__.py", line 130, in _get_pyproject_path
    return _find_higher_file("pyproject.toml", start=start)
  File "/Users/cfb/cisco/git/de/pistol.poetry/venv-pistol/lib/python3.9/site-packages/poetry_dynamic_versioning/__init__.py", line 122, in _find_higher_file
    for level in [start, *start.parents]:
AttributeError: 'LocalPath' object has no attribute 'parents'
```

pyproject.toml
```
[tool.poetry]
name = "pistol"
version = "0.0.0"
description = "Polaris Image Smithing ToOLs"
authors = [
    "Chet Burgess <cheburge@cisco.com>",
    "Darren Sanders <darrsand@cisco.com>"
]
classifiers = [
    "Development Status :: 4 - Beta",
    "Environment :: Console",
    "Intended Audience :: Developers",
    "Intended Audience :: Information Technology",
    "Intended Audience :: System Administrators",
    "Natural Language :: English",
    "Operating System :: POSIX :: Linux",
    "Programming Language :: Python :: 3",
    "Programming Language :: Python :: 3.6",
    "Programming Language :: Python :: 3.7",
    "Programming Language :: Python :: 3.8",
    "Programming Language :: Python :: 3.9",
    "Programming Language :: Python :: 3.10",
    "Topic :: System :: Systems Administration",
    "Topic :: Utilities",
]
exclude = ["pistol/tests/*"]

[tool.poetry.scripts]
get_version = 'pistol.cli.get_version:main'
get_remote_url = 'pistol.cli.get_remote_url:main'
get_repo_name = 'pistol.cli.get_repo_name:main'
git_list_tags = 'pistol.cli.git_list_tags:main'
find_best_tag = 'pistol.cli.find_best_tag:main'
find_best_version_dir = 'pistol.cli.find_best_version_dir:main'
tag_new_release = 'pistol.cli.tag_new_release:main'
render_templates = 'pistol.cli.render_templates:main'

[tool.poetry.dependencies]
python = ">=3.6.1,<4.0.0"
click = "^7.1.2"
pygit2 = "^1.5.0"
cached-property = "^1.5.2"
PyYAML = "^5.4.1"
terminaltables = "^3.1.0"
deepmerge = "^0.1.1"
Jinja2 = "^2.11.3"

[tool.poetry.dev-dependencies]
bandit = "^1.7.0"
coverage = "^5.5"
fixtures = "^3.0.0"
flake8 = "^3.8.4"
flake8-bugbear = "^20.11.1"
flake8-docstrings = "^1.5.0"
isort = "^5.7.0"
pyfakefs = "^4.4.0"
stestr = "^3.1.0"
testscenarios = "^0.5.0"
testresources = "^2.0.1"
testtools = "^2.4.0"

[tool.poetry-dynamic-versioning]
enable = true
vcs = "git"
style = "semver"
pattern = "^(?P<base>\\d+\\.\\d+\\.\\d+)$"
format-jinja = """
    {%- set pre = None -%}
    {%- set metadata = None -%}
    {%- set build_type = env['POERTY_BUILD_TYPE']|default('NONE') -%}
    {%- if distance != 0 -%}
        {%- set pre = ['dev', distance] -%}
        {%- if build_type == 'premerge' -%}
            {%- set metadata = [commit] -%}
        {%- endif -%}
    {%- endif -%}
    {{ serialize_semver(base, pre, metadata) }}
"""

[build-system]
requires = ["poetry-core>=1.0.0", "poetry-dynamic-versioning"]
build-backend = "poetry.core.masonry.api"
```

tox.ini
```
[tox]
isolated_build = true
envlist = py3, pep8, bandit, formatcheck
toxworkdir={toxinidir}/tox

[testenv]
require_locked_deps = true
install_dev_deps = true
passenv =
    BASEPYTHON
basepython = python3
envdir = {toxworkdir}/py3
setenv =
    VIRTUAL_ENV={envdir}
whitelist_externals =
    bash
    echo
    mkdir
commands =
    failing: stestr run --failing
    pep8: flake8 {posargs}
    debug: python -m testtools.run discover '{posargs}'
    test: python -m testtools.run '{posargs}'
    bandit: bandit -c bandit.yml -r pistol
    venv: {posargs}

[testenv:py3]
setenv =
    PYTHON=coverage run --source pistol --parallel-mode
commands =
    stestr run '{posargs}'
    stestr slowest
    coverage combine
    coverage report --fail-under=100

[testenv:formatcheck]
commands =
    isort --fss --sl -c pistol

[testenv:format]
commands =
    isort -fss -sl -rc pistol

[testenv:cover]
setenv =
    {[testenv]setenv}
    PYTHON=coverage run --source pistol --parallel-mode
commands =
    stestr run '{posargs}'
    coverage combine
    coverage report
    coverage xml -o coverage.xml
    mkdir -p cover
    coverage html -d cover
    echo "Point your browser at cover/index.html to see the HTML report."

[flake8]
max-line-length = 79
max-complexity = 13
doctests = True
filename= *.py
exclude =
    .git,
    .tox,
    tox,
    __pycache__,
    alembic,
    bin,
    lib,
    build,
    dist,
    .eggs
show-source = true
ignore =
    D10  # ignore missing docstrings (for now)
    D202 # No blank lines allowed after function docstring
    W503 # Conflicts with new W504

[coverage:run]
omit =
    *pistol/tests*

[isort]
line_length = 79
multi_line_output = 3
indent='    '
include_trailing_comma = True
use_parentheses = True
```

I'm happy to provide more info if necessary, and apologies if I'm doing something stupid/missed something obvious!

**Comment by cburgess** [comment]
I did some more debugging this morning and I traced the issue down to my use of `tox-poetry-installer` to manage poetry deps in tox (https://pypi.org/project/tox-poetry-installer/). If I don't load that plugin this I don't get the above stack trace. Also I went ahead and created a much simpler Hello World type project to show the issue. I've attached a tar.gz of the simple reproducible test cast. Just run the `run_test.sh` script to reproduce it. The script will build a venv in the test directory (it requires python and virtualenv be in your path some place) with all the tools it needs and then execute tox using the config in the test project.

[pvt.tar.gz](https://github.com/mtkennerly/poetry-dynamic-versioning/files/6091622/pvt.tar.gz)

**Comment by mtkennerly** [comment]
Thanks for all the detail here! I'll look into this.

**Comment by cburgess** [comment]
@mtkennerly I'm actually working on a patch now. Seems like a simply case where with `tox-poetry-installer` (and other tox poetry plugins) you get a string and not a pathlib object. Simple fix. I should have a PR up in a few minutes for you.

**Comment by cburgess** [comment]
I tested this fix with both `tox-poetry-installer 0.6.4` and `tox-poetry-dev-dependencies 0.0.9`. Both of them resulted in the same traceback as mentioned above without the patch, and both now seem to correctly function with the patch.

**Comment by mtkennerly** [comment]
I just published v0.12.4 with your fix 🎉

**Comment by cburgess** [comment]
Awesome thanks!
